iceoryx_posh/test/moduletests/test_popo_untyped_server.cpp

```diff
@@ -0,0 +1,179 @@
+
+    EXPECT_CALL(sut.mockPort, releaseRequest(_)).Times(0);
+
+    sut.releaseRequest(nullptr);
+}
+
+TEST_F(UntypedServer_test, LoanCallsCallsUnderlyingPortWithSuccessResult)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f39d58f3-b25e-4515-852d-c3afa5519e5a");
+
+    constexpr uint64_t PAYLOAD_SIZE{8U};
+    constexpr uint64_t PAYLOAD_ALIGNMENT{32U};
+    const iox::cxx::expected<ResponseHeader*, AllocationError> allocateResponseResult =
+        iox::cxx::success<ResponseHeader*>{responseMock.userHeader()};
+
+    EXPECT_CALL(sut.mockPort, allocateResponse(requestMock.userHeader(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT))
+        .WillOnce(Return(allocateResponseResult));
+
+    auto loanResult = sut.loan(requestMock.userHeader(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT);
+    ASSERT_FALSE(loanResult.has_error());
+    EXPECT_THAT(loanResult.value(), Eq(responseMock.sample()));
+}
+
+TEST_F(UntypedServer_test, LoanCallsCallsUnderlyingPortWithErrorResult)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d813b550-64b2-490f-a9f4-bafc9ddc7696");
+
+    constexpr uint64_t PAYLOAD_SIZE{8U};
+    constexpr uint64_t PAYLOAD_ALIGNMENT{32U};
+    constexpr AllocationError ALLOCATION_ERROR{AllocationError::RUNNING_OUT_OF_CHUNKS};
+    const iox::cxx::expected<ResponseHeader*, AllocationError> allocateResponseResult =
+        iox::cxx::error<AllocationError>{ALLOCATION_ERROR};
+
+    EXPECT_CALL(sut.mockPort, allocateResponse(requestMock.userHeader(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT))
+        .WillOnce(Return(allocateResponseResult));
+
+    auto loanResult = sut.loan(requestMock.userHeader(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT);
+    ASSERT_TRUE(loanResult.has_error());
+    EXPECT_THAT(loanResult.get_error(), Eq(ALLOCATION_ERROR));
+}
+
+TEST_F(UntypedServer_test, SendWithValidPayloadPointerCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "707cfdd8-05ae-490c-8cbb-9a4253135937");
+
+    EXPECT_CALL(sut.mockPort, sendResponse(responseMock.userHeader())).Times(1);
+
+    sut.send(responseMock.sample());
+}
+
+TEST_F(UntypedServer_test, SendWithNullpointerDoesNotCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d7552fc4-1e7c-40c2-b0b4-a24593dc0607");
+
+    EXPECT_CALL(sut.mockPort, sendResponse(_)).Times(0);
+
+    sut.send(nullptr);
+}
+
+TEST_F(UntypedServer_test, ReleaseResponseWithValidPayloadPointerCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "db4be6c4-fe6f-439c-b1c7-875909dd9573");
+
+    EXPECT_CALL(sut.mockPort, releaseResponse(responseMock.userHeader())).Times(1);
+
+    sut.releaseResponse(responseMock.sample());
+}
+
+TEST_F(UntypedServer_test, ReleaseResponseWithNullpointerDoesNotCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "63cdadf3-5745-4196-a5dc-5b8925c14289");
+
+    EXPECT_CALL(sut.mockPort, releaseResponse(_)).Times(0);
+
+    sut.releaseResponse(nullptr);
+}
+
+} // namespace
```

Comment:
Again, large part of the test code is duplicated. Could be combined into a `TYPED_TEST` with `test_popo_untyped_client.cpp`.